runners/direct-java/src/main/java/org/apache/beam/runners/direct/DoFnLifecycleManager.java

```diff
@@ -58,7 +59,7 @@ private DoFnLifecycleManager(DoFn<?, ?> original) {
 
   public DoFn<?, ?> get() throws Exception {
     Thread currentThread = Thread.currentThread();
-    return outstanding.get(currentThread);
+    return outstanding.get(currentThread).getDoFn();
```

Comment:
Where is this called from? My sense is that if we can avoid the need to pull out the original `DoFn` then it is pretty clean.